crates/ruff_linter/src/linter.rs

```diff
@@ -346,7 +346,7 @@ pub fn check_path(
             &directives.noqa_line_for,
             parsed.has_valid_syntax(),
             settings,
-            &suppressions,
+            &mut suppressions,
```

Comment:
Should we just pass `suppressions` here (the owned value), given that they aren't used afterwards?


Another alternative would be to store them on `LintContext` (I don't remember if that would be hard. It's okay to only look at this at the very end. Just floating this as an idea). Ultimately, I think, where we store it probably depends on where we want to use it, e.g. e might need to pass them to a new lint rule that checks for unclosed suppressions)